### PR TITLE
Rearrange past events

### PR DESCRIPTION
**Pull request by madhurabhujbal**
Ticket link: https://github.com/isaaccomputerscience/isaac-cs-issues/issues/666

This pull request changes how "My booked events" are sorted on the Events page:

- Upcoming (future) events are now listed first, with the soonest event at the top.
- Cancelled and past events are placed at the bottom, with the most recent past events listed first.
- The sorting ensures cancelled events always appear last.
- The logic is updated to handle missing or invalid event dates gracefully.
- New tests are added to verify the correct order of events, including edge cases like identical dates or cancelled events.

### CODE REVIEW

File: src/mocks/data.ts [hunk 1]
```diff
@@ -874,6 +874,8 @@ export const mockEventBooking: DetailedEventBookingDTO = {
   updated: 1695897589235,
 };
 
+// Add these after the existing mockEvent definition
```
Comment (Parth097) [inline src/mocks/data.ts:877]: What is the purpose of this comment??

Comment (madhurabhujbal) [reply src/mocks/data.ts:877]: Not really sure. It was there already.

File: src/app/components/pages/Events.tsx [hunk 2]
```diff
@@ -164,11 +164,26 @@ export const Events = withRouter(({ history, location }: RouteComponentProps) =>
                 <div className="my-4">
                   <Row>
                     {(statusFilter === EventStatusFilter["My booked events"]
-                      ? events.sort((bookedEvent1, bookedEvent2) => {
-                          const oldEventDate = bookedEvent1.date ? new Date(bookedEvent1.date).getTime() : 0;
-                          const newEventDate = bookedEvent2.date ? new Date(bookedEvent2.date).getTime() : 0;
-                          // The return value of the subtraction below will always be positive as the timestamp for bookedEvent2 will be bigger than of bookedEvent1. This will sort the events in descending order i.e newest first
-                          return newEventDate - oldEventDate;
+                      ? events.sort((a, b): number => {
```
Comment (Parth097) [inline src/app/components/pages/Events.tsx:167]: There is a lot of in-line logic here which can be separated out into a helper function. This can be done at the top of the file. 
Once the logic is separated, we can call the function here instead of the in-line logic.

File: src/mocks/data.ts (hunk 1 above)
Comment (Parth097) [reply src/mocks/data.ts:877]: So why is it coming as your changes??

Comment (madhurabhujbal) [reply src/mocks/data.ts]: I've removed it

File: src/app/components/pages/Events.tsx (hunk 2 above)
Comment (madhurabhujbal) [reply src/app/components/pages/Events.tsx]: Extracted the logic into a helper function

**Review by Parth097 — APPROVED** [review]
